modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -229,6 +235,9 @@ public void register(ConfigurationStorage configurationStorage) throws Configura
      *      storage.
      */
     public void initialize(ConfigurationType storageType) throws ConfigurationValidationException, ConfigurationChangeException {
+        if (bootstrapped)
```

Comment:
I am not convinced that this is a correct behavior

Reply:
Got rid of this flag as you recommend.